ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Genesis/Governor.hs

```diff
@@ -239,16 +240,35 @@ updateLoEFragGenesis cfg tracer getCandidates getHandles =
           cschLatestSlot (handles Map.! peer)
       pure (candidateSuffixes, handles, theirTips, latestSlots, loeFrag)
 
-    let
-      -- TODO: use a forecasted ledger view for the intersection
-      -- slot (tip of LoE frag).
-      -- REVIEW: Have we talked about this?
-      sgen = computeGenesisWindow (configLedger cfg) (ledgerState immutableLedgerSt)
-      (losingPeers, bounds) =
-        densityDisconnect sgen (configSecurityParam cfg) candidateSuffixes theirTips latestSlots loeFrag
-      loeHead = AF.headAnchor loeFrag
+    let msgen :: Maybe GenesisWindow
+        msgen = eitherToMaybe $ runQuery qry summary
+          where
+            -- Usually, the initial ledger state (with tip slot 'Origin') is in
+            -- the first non-empty era, see 'injectInitialExtLedgerState' and
+            -- its call to 'State.extendToSlot' with @'SlotNo' 0@.
+            slot    = fromWithOrigin 0 $ AF.headSlot loeFrag
```

Comment:
I don't understand the comment. Is it explaining why it's fine to not make the distinction between `Origin` and `NotOrigin (SlotNo 0)`?

Reply:
> Is it explaining why it's fine to not make the distinction between `Origin` and `NotOrigin (SlotNo 0)`?

Now that I think about it again, an alternative would be to use `succWithOrigin` here and argue that we should use the Genesis window associated with the first slot *after* the intersection. This only makes a difference with the current behavior if we the intersection is in the last slot of an era. In that case, all slots in the Genesis window are in the successor era, so `succWithOrigin` actually sounds nicer here. Will change it to that :+1:

Reply:
6059c08a1d89e6c35ce0144f52f5dea2c6b423f7